### ISSUE TITLE
Unit tests

### ISSUE THREAD

**Issue by dselans** [9corp/9volt#23]
Once baseline functionality is in place, we can start working out unit tests.

**Comment by dselans** [comment]
This will probably take a while; progress is tracked in #76

**Comment by dselans** [comment]
Putting this back in the backlog for the time being - lots of work still needed.

**Comment by dselans** [comment]
So *NUMBER 1* thing that needs to be unit tested is the director check distribution logic. That is a heavy piece of machinery and we **must be certain** that it works after every single commit.

**Comment by jessedearing** [comment]
What's needed here. Starting with director, I see a lot of spending specs. Should I just make those pass?

**Comment by dselans** [comment]
@jessedearing So I ended up writing out *a lot* of the test cases and left the implementation for later. The director *absolutely* needs to be worked on, but I am right now in the middle of redoing some bits and pieces of it (and the `dal` as well). I believe @talpert is working on testing the cluster component, so basically anything else is up for grabs!

The most important pieces needing testing in order of importance: cluster, director, manager, alerter, monitor, checks, state, event. Pick and choose whichever one - maybe create a separate issue and stick into the projects board so everyone knows what you're on?

**Comment by dselans** [comment]
This is a shared task essentially; if folks want to split it out into separate notes (or separate issues) please feel free @jessedearing @talpert @relistan.

**Comment by talpert** [comment]
I created a note and moved it in progress.

**Comment by dselans** [comment]
I've written a bunch of tests for overwatch and cluster; will continue with director and so forth now that the logging testing story is a little more fleshed out.
